### PR TITLE
Make deprecation notice of the spam checker doc more obvious

### LINKED ISSUE #10333: Deprecation notice on the spam checker doc isn't obvious enough
Looks like the deprecation notice on https://matrix-org.github.io/synapse/latest/spam_checker.html is easier to miss than I thought. @anoadragon453 suggested:

> as far as the warning goes, I would start by putting it below "Handling spam in Synapse", then probably bumping it to a higher header number (maybe even just h1).
>
> we can do red - none of the themes have red text as the default (yet)

### PR DESCRIPTION
**Pull request by babolivier**
Old:

image

New:

image

Fixes #10333

### CODE REVIEW

**Comment by babolivier** [comment]
PRing against master on purpose so it reaches the stable docs without having to wait for a full release cycle

**Comment by richvdh** [comment]
(please don't forget to merge back into develop after merging to master)

File: docs/spam_checker.md
Comment (anoadragon453) [inline docs/spam_checker.md:4]: Curious that markdown links don't work within custom HTML tags - but I guess that's probably a markdown thing.

**Review by anoadragon453 — APPROVED** [review]
Looking good!

**Pull request merged**
